nimrod/report/report_analysis.py

```diff
@@ -0,0 +1,170 @@
+                cells = line.split(",")
+                object_map = {cells[0]: [cells[1], cells[2], 0, 0]}
+                objects_data.update(object_map)
+        except:
+            print("Error on objects_report.csv in " + str(path_suite_original))
+            return {}
+
+        try:
+            objects_report_file = open(path_suite_modified + "/objects_report.csv")
+            objects_report = objects_report_file.read()
+            objects_report_file.close()
+
+            lines = objects_report.split("\n")
+            for line in lines[1:-1]: # Exclude the first one because contains the headers and the last one because it is a empty line
+                cells = line.split(",")
+                if cells[0] in objects_data: # If the class is already in the dictionary, just change the value in the list
+                    objects_data.get(cells[0])[2] = cells[1]
+                    objects_data.get(cells[0])[3] = cells[2]
+                else:
+                    object_map = {cells[0]: [0, 0, cells[1], cells[2]]}
+                    objects_data.update(object_map)
+        except:
+            print("Error on methods_report.csv in " + str(path_suite_modified))
+            return {}
+
+        return objects_data
+
+    def write_comparison(self, path_suite_original, path_suite_modified, methods, objects):
+        suites_path = path_suite_original[:path_suite_original.rfind("/")]  # Get the directory where the test suites are
+        reports_path = suites_path + "/reports"
+
+        if os.path.isdir(reports_path) is False:
+            os.mkdir(reports_path)
+
+        suite_name_original = path_suite_original[path_suite_original.rfind("/") + 1:]
+        suite_name_modified = path_suite_modified[path_suite_modified.rfind("/") + 1:]
+
+        with open(reports_path + "/methods_report_" + suite_name_original + "_" + suite_name_modified + ".csv", "w") as output_file:
+            headers = "Methods called,Number of times(Original),Number of times(Modified),Modified - Original,Percentage(Gain/Loss),\n"
+            output_file.write(headers)
+            for key in methods:
+                number_original = int(methods.get(key)[0])
+                number_modified = int(methods.get(key)[1])
+                sub = number_modified - number_original
+
+                if number_original == 0:
+                    relative = "INF"
+                elif number_modified == 0:
+                    relative = "-INF"
+                else:
+                    relative = (sub / number_original) * 100
+                    relative = str(round(relative, 2)).replace(".", ",")
+
+                text = key + "," + str(number_original) + "," + str(number_modified) + "," + str(sub) + "," + "\"" + relative + "\"%" + "\n"
+                output_file.write(text)
+
+            output_file.close()
+
+        with open(reports_path + "/objects_report_" + suite_name_original + "_" + suite_name_modified + ".csv", "w") as output_file:
```

Comment:
A previous comment regarding duplicated code that I made can be applied here.
This duplicated code could be eliminated by extracting these duplicated lines as a new method.
I know that here the file headers are different, but this information could be received as a parameter in the new proposed extracted method.